lib/ui/figma_design.dart

```diff
@@ -0,0 +1,233 @@
+
+class Figma {
+  // Colors
+  static const Colors colors = Colors();
+
+  // Typography
+  static Typography typo = Typography();
+
+  // Buttons
+  static Buttons buttons = Buttons();
+
+  // Icons
+  static Icons icons = Icons();
+}
+
+class Colors {
+  const Colors();
+
+  final Color backgroundColor = const Color(0xff181305);
+  final Color primaryColor = const Color(0xffFFBA00);
+  final Color secondaryColor = const Color(0xffF9F2E0);
+  final Color errorColor = const Color(0x7fFF0000);
+
+  final Color disabledColor = const Color(0xffD9D9D9);
+  final Color primaryButtonHoverColor = const Color(0xffFFDA76);
+  final Color secondaryButtonHoverColor = const Color(0xff5F490E);
+  final Color navbarSelectedBackgroundColor = const Color(0x99745501);
+
+  final Color transparentColor = const Color(0x00000000);
+}
+
+class Typography {
+  Typography();
+
+  final TextStyle header1 = const TextStyle(
+    fontFamily: "Oswald",
+    fontWeight: FontWeight.bold,
+    fontSize: 40,
+    letterSpacing: -0.2,
+  );
+
+  final TextStyle header2 = const TextStyle(
+    fontFamily: "Oswald",
+    fontWeight: FontWeight.bold,
+    fontSize: 24,
+    letterSpacing: -0.2,
+  );
+
+  final TextStyle body = const TextStyle(
+    fontFamily: "Lato",
+    fontWeight: FontWeight.normal,
+    fontSize: 16,
+    height: 1.4,
+  );
+
+  final TextStyle bold = const TextStyle(
+    fontFamily: "Lato",
+    fontWeight: FontWeight.bold,
+    fontSize: 16,
+    height: 1.4,
+  );
+
+  final TextStyle smallerText = const TextStyle(
+    fontFamily: "Lato",
+    fontWeight: FontWeight.normal,
+    fontSize: 14,
+  );
+
+  final TextStyle preTitle = const TextStyle(
+    fontFamily: "Lato",
+    fontWeight: FontWeight.bold,
+    fontSize: 10,
+    letterSpacing: 0.3,
+  );
+
+  final TextStyle button = const TextStyle(
+    fontFamily: "Lato",
+    fontWeight: FontWeight.bold,
+    fontSize: 10,
+    letterSpacing: 0.3,
+  );
+}
+
+class Buttons {
+  Buttons();
+
+  final ButtonStyle primaryButtonStyle = ButtonStyle(
+    textStyle: MaterialStateProperty.all(Figma.typo.button),
+    fixedSize: MaterialStateProperty.all(const Size(double.infinity, 35)),
+    // To size the button, wrap it with a SizedBox with a width
+    foregroundColor: MaterialStateProperty.resolveWith<Color?>(
+      // Typically the color of the text on the button
+      (Set<MaterialState> states) {
+        if (states.contains(MaterialState.error)) {
+          return Figma.colors.errorColor;
+        }
+        return Figma.colors.backgroundColor;
+      },
+    ),
+    backgroundColor: MaterialStateProperty.resolveWith<Color?>(
+      // The color of the button itself
+      (Set<MaterialState> states) {
+        if (states.contains(MaterialState.disabled)) {
+          return Figma.colors.disabledColor;
+        }
+        if (states.contains(MaterialState.error)) {
+          return Figma.colors.errorColor;
+        }
+        return Figma.colors.primaryColor;
```

Comment:
have you tried this with a `switch-case-default`? Where there are more conditions than a regular `if-else`, then I'd suggest a `switch-case` approach due to its readability

Reply:
the same comment would go for the rest of the multi-if cases

Reply:
nevermind, I checked and it is not possible in this scenario